packages/js/src/ui/components/Notification/DefaultNotification.tsx

```diff
@@ -14,20 +16,49 @@ const NotificationBody = (props: NotificationBodyProps) => {
 };
 
 type DefaultNotificationProps = {
-  notification: InboxNotification;
+  notification: Notification;
+  onNotificationClick?: NotificationClickHandler;
+  onActionClick?: NotificationActionClickHandler;
 };
 
 //TODO: Complete the implementation
 export const DefaultNotification = (props: DefaultNotificationProps) => {
   const style = useStyle();
   const { t, locale } = useLocalization();
 
+  const handleNotificationClick: JSX.EventHandlerUnion<HTMLAnchorElement, MouseEvent> = (e) => {
+    e.stopPropagation();
+    e.preventDefault();
+
+    if (!props.notification.isRead) {
+      props.notification.read();
+    }
+
+    props.onNotificationClick?.({ notification: props.notification });
+    if (props.notification.redirect?.url) {
+      window.open(props.notification.redirect?.url, '_blank', 'noreferrer noopener');
+    }
+  };
+
+  const handleActionButtonClick = (action: ActionTypeEnum, e: MouseEvent) => {
+    e.stopPropagation();
+
+    if (action === ActionTypeEnum.PRIMARY) {
+      props.notification.completePrimary();
+    } else {
+      props.notification.completeSecondary();
+    }
+
+    props.onActionClick?.({ notification: props.notification, action });
```

Comment:
when action button is clicked:
- complete the action
- call the `onActionClick` prop